lib/services/role.go

```diff
@@ -2034,6 +2034,37 @@ type kubernetesClusterLabelMatcher struct {
 	clusterLabels map[string]string
 }
 
+// KubernetesResourceMatcher matches a role against a Kubernetes Resource.
+// Kind is must be stricly equal but namespace and name allow wildcards.
+type KubernetesResourceMatcher struct {
+	types.KubernetesResource
```

Comment:
Do we need the matcher to expose the same interface as KubernetesResource? Ie, do we need the inlining?

Could we use a private, named field instead?